### PR TITLE
Optimize long route and collection performance

### LINKED ISSUE #18: Optimize long route and collection performance
## Context

Performance review focused on long ultra-distance routes and route collections. The app's current architecture is feature-friendly, but several paths scale directly with full `RoutePoint[]` size. Collections amplify this because selected segments are loaded, stitched into a second full array, and then reused by map, elevation, ETA, weather, POI, and climb consumers.

## Main Concerns

- Startup and tab mount can load too many route points. `app/_layout.tsx` prefetches `loadRoutesAndPoints()`, and `store/routeStore.ts` loads points for all visible routes even though the map renders only the active route or active collection segments.
- Collection activation duplicates large arrays. `services/stitchingService.ts` loads selected segment points and creates a new stitched point object for every raw point, while `store/collectionStore.ts` keeps both stitched points and per-segment raw points in memory.
- Map rendering sends full-resolution route geometry to Mapbox. `components/map/RouteLayer.tsx` converts every route point into GeoJSON coordinates through `utils/geo.ts:routeToGeoJSON`.
- Snapping and ETA lookups use linear scans. `findNearestPointOnRoute()` scans every point, `getETAToDistance()` scans forward from the current index, and starred POI ETA does another per-POI scan.
- POI collection display loads and stitches all POIs for all active segments, then filters/sorts in JS. This will get heavier on long collections with many cached POIs.
- Offline POI processing associates every candidate POI by scanning route geometry, which can become `candidate POIs × route points`.
- Collection detail is a heavy planning screen: it stitches selected segments, loads unselected variant points, renders mini-map routes, renders a full collection elevation profile, and recomputes per-segment ETA totals.

## Proposed Work

1. Change app startup to metadata-only route loading, then lazy-load points for the active route/collection and detail screens.
2. Add simplified/cached route geometry for Mapbox rendering so full GPX precision is not sent to the map layer.
3. Replace distance-based linear scans with binary search helpers.
4. Improve snapping with a spatial/grid index or a local search window around the last snapped point.
5. Window POI queries by route distance for the common riding view, while preserving full-list behavior when explicitly expanded.
6. Rework collection stitching into a lighter view model using segment offsets and raw point references where possible.
7. Cache per-route derived data such as ETA totals, simplified map geometry, and possibly cumulative ETA arrays.
8. Add performance-oriented tests/benchmarks or lightweight instrumentation for representative long routes and multi-segment collections.

## Acceptance Criteria

- Opening the app with multiple imported long routes does not eagerly load every visible route's full point array.
- Activating a long collection avoids unnecessary duplicate full-resolution arrays where downstream consumers can work from segment offsets or simplified geometry.
- Map rendering uses simplified route geometry without degrading the route's visual usefulness.
- ETA-to-distance lookups use binary search or equivalent sub-linear lookup.
- POI list/map/elevation consumers avoid repeatedly copying and sorting full collection POI lists during ordinary riding interactions.
- Add notes or measurements showing before/after behavior for a long route or stitched collection scenario.

### PR DESCRIPTION
**Pull request by conqeror**
## Summary

- Load route and collection metadata during startup, then lazy-load full route geometry for the active riding/detail contexts.
- Keep active collection state lighter by omitting retained raw segment point arrays, rendering collections from stitched geometry, and avoiding route-store injection for collection detail mini maps.
- Add cached simplified Mapbox route geometry, binary distance helpers for ETA/slices, local-window route snapping, windowed POI map rendering, and a grid-backed POI association path.
- Add performance notes plus tests for stitching windows, route snapping fallback, map simplification/cache behavior, distance boundary helpers, and spatial POI association.

Fixes #18

## Verification

- `npx tsc --noEmit`
- `npm test`
- `npm run lint`
- `git diff --check`
- Commit hook: `oxfmt`, `oxlint --fix`, `tsc --noEmit`

### CODE REVIEW

File: components/map/MapView.tsx [hunk 1]
```diff
@@ -375,11 +399,20 @@ export default function MapScreen() {
             />
           );
         })}
+        {activeCollectionRoute && activeRoutePoints && activeRoutePoints.length > 1 && (
+          <RouteLayer
+            key={`${activeCollectionRoute.id}-${mapStyle.styleKey}`}
+            route={activeCollectionRoute}
+            points={activeRoutePoints}
```
[the remark this answers is not in the document]
Comment (conqeror) [reply components/map/MapView.tsx:406]: @codex, this is fine, in reality there won't be big gaps and connecting them with line is fine behaviour
